spec/controllers/users_controller_spec.rb

```diff
@@ -91,39 +128,60 @@
   end
 
   describe "GET #change_username" do
+    let(:user) { create(:user) }
+    let(:success) { expect(response).to render_template(:change_username) }
     subject { get :change_username, params: { id: user } }
 
     context "when logged in as a valid user" do
-      let(:user) { create(:user) }
-
       before do
         fake_login_known_user(user)
       end
 
       it "shows the change username form" do
         subject
-        expect(response).to render_template(:change_username)
+        success
       end
     end
 
+    it_behaves_like "an action only authorized admins can access", authorized_roles: %w[superadmin policy_and_abuse]
     it_behaves_like "blocks access for banned and suspended users"
   end
 
   describe "POST #changed_username" do
+    let(:user) { create(:user) }
+    let(:new_username) { "foo1234" }
+    let(:success) do
+      expect(user.reload.login).to eq(new_username)
+    end
+
     subject do
-      post :changed_username, params: { id: user, new_login: "foo1234", password: user.password }
+      post :changed_username, params: { id: user, new_login: new_username, password: user.password }
     end
 
     context "when logged in as a valid user" do
-      let(:user) { create(:user) }
-
       before do
         fake_login_known_user(user)
       end
 
       it "updates the user's username" do
-        expect { subject }
-          .to change { user.reload.login }
+        subject
+        success
+      end
+    end
+
+    it_behaves_like "an action only authorized admins can access", authorized_roles: %w[superadmin policy_and_abuse] do
+      let(:user) { create(:user, banned: true) }
```

Comment:
Why a banned user?

Reply:
Non-exhaustive checking that users who cannot access this page themselves can be renamed by an admin